src/document.h

```diff
@@ -1435,6 +1432,28 @@ struct Document {
                 }
                 return nullptr;
 
+            case A_COLLAPSE: {
+                if (selected.xs * selected.ys == 1) return _(L"More than one cell must be selected.");
+                auto fc = selected.GetFirst();
+                if (!fc->HasContent()) return _(L"The first cell in selection must have content.");

```

Comment:
Why is this? even if this cell has no content, aren't you adding the text of all other cells to it below?
What I meant was I wasn't sure if `MultiCellDeleteSub` keeps this cell, but I guess it does?

Reply:
The problem arises when you have cells in the selection that all have no content and the selection is surrounded by cells without content. Then MultiCellDeleteSub will also delete the row/column of the first cell, leading to a crash...

Reply:
You are right we can check after collecting the text. If there is still no content, then leave the routine...